ai/src/ai/offline_common/golden_dataset.py

```diff
@@ -1,9 +1,31 @@
-from ai.common.example import GoldenExample
-from ai.common.prompt_context import PromptContext
+import json
+import os
 
+from ai.common.entity_store import get_data_path
+from ai.common.example import golden_example_store
+from ai.common.executor import ProducerExecutor
 
-def create_golden_dataset(
-  examples: list[GoldenExample], prompt_context: PromptContext
-) -> str:
-  print("create_golden_dataset", examples, prompt_context)
-  return "not yet implemented"
+
+def create_golden_dataset(*, producer_id: str, dataset_name: str) -> str:
+  #  examples: list[GoldenExample],
+  golden_dataset_path = os.path.join(
+    get_data_path("golden_datasets"), f"{dataset_name}.jsonl"
+  )
+  with open(golden_dataset_path, "w") as f:
+    examples = golden_example_store.get_all()
+    producer_executor = ProducerExecutor(producer_id)
+    for example in examples:
+      messages = producer_executor.get_provider_executor().format_messages(
+        example.input
+      )
+      if example.output.udiff_output is None:
```

Comment:
Why is udiff output required?

Reply:
we should make it configurable - I've updated: https://github.com/google/mesop/issues/935; filed https://github.com/google/mesop/issues/959 to improve our golden output story